File: EasyDataAPIWrapper.Java/src/main/java/services/ApiClientBuilder.java
package services;

public class ApiClientBuilder {
    private ApiClient _apiClient = new ApiClient();

    public static ApiClientBuilder init()
    {
        return new ApiClientBuilder();
    }

    public ApiClient build() {
     return  _apiClient;
    }

    public ApiClientBuilder setHost(String host)
    {
        _apiClient.setHost(host);
        return this;
    }

    public ApiClientBuilder setPort(int port)
    {
        _apiClient.setPort(String.valueOf(port));
        return this;
    }

    public ApiClientBuilder setProtocol(String protocol)
    {
        _apiClient.setProtocol(protocol);
        return this;
    }

    public ApiClientBuilder useHttps()
    {
        _apiClient.setProtocol("https");
        return this;
    }

    public ApiClientBuilder setBaseUrl(String url)
    {
        _apiClient.setBaseUrl(url);
        return this;
    }

    public ApiClientBuilder setDateFormat(String format)
    {
        _apiClient.setDateFormat(format);
        return this;
    }

    public ApiClientBuilder setJsonCharset(String charset)
    {
        _apiClient.setJsonCharset(charset);
        return this;
    }
}

File: EasyDataApiWrapper.C#/EasyDataApiWrapper/services/ApiClientBuilder.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EasyDataApiWrapper.services
{
    public class ApiClientBuilder
    {
        private readonly ApiClient _apiClient = new ApiClient();

        public ApiClientBuilder Init()
        {
            return new ApiClientBuilder();
        }

        public ApiClient Build() => _apiClient;

        public ApiClientBuilder SetHost(string host)
        {
            _apiClient.Host = host;
            return this;
        }

        public ApiClientBuilder SetPort(int port)
        {
            _apiClient.Port = port.ToString();
            return this;
        }

        public ApiClientBuilder SetProtocol(string protocol)
        {
            _apiClient.Protocol = protocol;
            return this;
        }

        public ApiClientBuilder UseHttps()
        {
            _apiClient.Protocol = "https";
            return this;
        }

        public ApiClientBuilder SetBaseUrl(string url)
        {
            _apiClient.BaseUrl = url;
            return this;
        }

        public ApiClientBuilder SetDateFormat(DateTimeZoneHandling format)
        {
            _apiClient.DateTimeZoneHandling = format;
            return this;
        }

        public ApiClientBuilder SetJsonCharset(string charset)
        {
            _apiClient.JsonCharset = charset;
            return this;
        }
    }
}
